### ISSUE TITLE
Building with Visual Studio does not work if Scons is not on the path

### ISSUE THREAD

**Issue by 0x0ACB** [godotengine/godot#96992]
### Tested versions

74de05a01c8716a42d4e3427f607d7bea76b35e5

### System information

Windows 11

### Issue description

The solution generated with `scons platform=windows vsproj=yes dev_build=yes` will not build unless `scons` is available on the path. I am using conda environments with python as is recommended. I only have scons installed in one of the virtual environments. Building normally with scons this way works fine, however if I generate an sln file and try to build it using Visual Studio the build fails because it can't find scons. Adding scons to the path solves the issue but that is more a workaround than an actual solution.

### Steps to reproduce

- run `scons platform=windows vsproj=yes dev_build=yes` in a conda venv
- try to build

### Minimal reproduction project (MRP)

-

**Comment by WhalesState** [comment]
Check the `Python Environments` window by clicking on `View` -> `Other Windows` -> `Python Environments`.

**Comment by 0x0ACB** [comment]
Does not work. It shows my environment and I set it as default. But I am still getting 
```
1>[vcvarsall.bat] Environment initialized for: 'x86_x64'
1>'scons' is not recognized as an internal or external command,
```

**Comment by WhalesState** [comment]
Check your system path variables, sort or add your current python environment on top of all other environments, then click ok and restart. you also should include `Python3x/Scripts` path.

![image](https://github.com/user-attachments/assets/f334bae7-6d36-43be-a28d-a1f60960c431)

**Comment by 0x0ACB** [comment]
Like I said in my initial post. That is a workaround to the issue yes, but its not a solution. The necessary paths should be added to the build command when the solution file is generated same way as what CMake does for those situations. I do not want to pollute my PATH with all kinds of python variables.

**Comment by WhalesState** [comment]
It's more related to system and how you install python, it should be documented, also another work around is to build with the full path of scons if you don't want to change the default python path system wide.

**Comment by 0x0ACB** [comment]
In CMake I can do something like this:
```
find_package(Python3 REQUIRED)
find_program(SCONS_PROGRAM scons HINTS "${Python3_STDLIB}/../Scripts/" REQUIRED)
set(GODOT_CMD cmake -E env BUILD_NAME=gw2cc SCRIPT_AES256_ENCRYPTION_KEY=${GODOT_SCRIPT_KEY} 
    ${SCONS_PROGRAM} 
        lto=none#${GODOT_LTO} 
        debug_symbols=yes 
        use_static_cpp=no 
        tests=no
        #d3d12=${GODOT_D3D12} mesa_libs="${GODOT_NIR_FOLDER}"
        platform=${GODOT_PLATFORM} 
        -j${N} 
        target=${GODOT_TARGET} 
        windows_subsystem=console
)
```
This way python only needs to be on the path during project generation. The build script embedded in the solution file will use absolute paths. This is a much better solution than forcing people to either put scons on the path globally or always run the build command manually. Your last solution also kinda prevents you from debugging with Visual Studio unless you also turn of Building before debugging.

**Comment by WhalesState** [comment]
I don't think it will cause issues with debugging, since scons generated files are located inside the godot dir and the command `scons` is only used once while building.

**Comment by 0x0ACB** [comment]
Visual Studio will by default always try to build the project before debugging it. Which will then fail because scons is not on the path. It is also inconvenient to have to first manually build and then hit debug. It's also easy to forget and then you are debugging a program that does not match you code.

**Comment by WhalesState** [comment]
I mean you change the command `scons` to the full path of scons, like in VSCode i can change it for all the tasks that runs before debugging, this will not search the system variables for scons path anymore.
![image](https://github.com/user-attachments/assets/15861b3f-396e-404a-a5e8-de81d44e998f)

**Comment by 0x0ACB** [comment]
Yes, also not really possible for Visual Studio. Scons regenerates the build script each time you rebuild. So you would need to completely edit the build script to change all of that. And then you also need to edit it again each time you want to add new source files.
